parser/media/src/main/java/de/danoeh/antennapod/parser/media/vorbis/VorbisInputStream.java

```diff
@@ -0,0 +1,85 @@
+package de.danoeh.antennapod.parser.media.vorbis;
+
+import org.apache.commons.io.IOUtils;
+
+import java.io.BufferedInputStream;
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayDeque;
+import java.util.Arrays;
+import java.util.Deque;
+
+
+public class VorbisInputStream extends FilterInputStream {
+    // capture_pattern + stream_structure_version
+    private static final int[] REST_PATTERN = {'g', 'g', 'S', 0};
+    private static final int HEADER_SKIP_LENGTH = 1 + 8 + 4 + 4 + 4;
+
+    private final Deque<Integer> buffer = new ArrayDeque<>();
+    private final BufferedInputStream bis;
+
+
+    protected VorbisInputStream(InputStream in) {
+        super(in);
+        this.bis = new BufferedInputStream(in);
+    }
+
+
+    private int readWithBuffer() throws IOException {
+        if (!buffer.isEmpty()) {
+            return buffer.removeFirst();
+        } else {
+            return this.bis.read();
+        }
+    }
+
+
+    @Override
+    public int read() throws IOException {
+        int readByte = readWithBuffer();
+
+        if (readByte == 'O') {
```

Comment:
Doesn't this break if anything inside the comments contains `OggS`? For example if a chapter title happens to be named after this? I think I would rather parse the length of the page from the header. Then `read()` can return the next character as long as we are below the page size. Then after the page, we parse the next header and updates the page length.